### ISSUE TITLE
Too many spurious guest accounts being created 

### ISSUE THREAD

**Issue by melanie-exygy** [civiform/civiform#5542]
**Describe the bug**
A lot of empty guest accounts are created with our current account allocation design.  This is causing a lot of clutter in the database.

**To Reproduce**

1. Go to the landing page.  Note that a default guest account has been created (see the account id in the url)
2. Log in and log out again.
3. Note that a different guest account is created.

**Expected behavior**
Guest accounts should not be created by default when a user arrives on the landing page.  A guest account should only be created when the user opts out of the login flow.  

**Additional context**
This may be a fairly large refactoring of the landing page given that it isn't set up to be able to show anything (including the list of available programs) without a user tied to it right now.

**Comment by bion** [comment]
This is WAI

**Comment by bion** [comment]
We used to have a landing page that gave applicants the option to log in or continue as guests, research showed less engagement from applicants with that page so we removed it in favor of immediately showing the list of programs. The code that loads and displays the programs assumes an account of some type exists (including guest accounts), and refactoring it would be a fairly big effort so we went with creating an account on initial pageload if necessary. We checked with Seattle to get a sense of traffic and how much of an issue it would be for the database size and determined it wouldn't be an issue. There's an issue for writing a recurring job for deleting the old accounts but there's not much motivating action on it at the moment: https://github.com/civiform/civiform/issues/4719
